Polyfills/Window/Source/Window.cpp

```diff
@@ -32,46 +40,85 @@ namespace
             , m_thread{&TimeoutDispatcher::WaitThenCallProc, this}
         { }
 
-        void Dispatch(std::shared_ptr<Napi::FunctionReference> func, std::chrono::system_clock::time_point time) 
-        {   
-            if (time <= std::chrono::system_clock::now())
+        TimeoutId Dispatch(std::shared_ptr<Napi::FunctionReference> func, Milliseconds time)
+        {
+            if (time <= 0)
             {
                 CallFunction(std::move(func));
-                return;
+                return 0;
             }
 
+            const auto now = Now();
             std::unique_lock<std::mutex> lk(m_mutex);
             const auto soonestTimePoint =
-                m_queue.empty()
-                    ? time + std::chrono::milliseconds{1}
-                    : m_queue.top().TimePoint;
+                m_timeMap.empty()
+                ? now + std::chrono::milliseconds{ 1 }
+            : (*m_timeMap.cbegin()).second.front().timePoint;
```

Comment:
I don't understand this variable. What does it mean to be the soonest time point?

Reply:
This is Robert's name for it. It's the time point that will be reached next. It's being used to tell the thread how long to sleep, i.e., "wake up the thread when the soonest time point is reached".

Reply:
Maybe change it to "earliestTimePoint"?

Reply:
Why is it 1 millisecond from now when the time map is empty?

Reply:
Should probably be `std::chrono::millisecond::max`?

Reply:
Oh wait, I think the way Robert set this up is the 1 millisecond setting wakes up the thread and the thread sets process function sets the next wake up time. This can probably be done better.

Reply:
Ya, good catch. I tested it with a single 1 second timeout and it never fires. Changing it to `TimePoint::max()` fixes it.